app/assets/javascripts/common.js

```diff
@@ -0,0 +1,112 @@
+/* ===================================== */
+/* =============== Alerts ============== */
+/* ===================================== */
+/* ===== Make Inline Alert ===== */
+function makeAlert(type, element_within, message){
+  if (element_within == ''){
+    element_within = '#main .alert-container:first';
+  }
+  // NOTE - this shall only be used to generate messages on pages that are not going to be reloaded. If you wish to reload the page AND have an alert when it loads, alert needs to be generated from backend.
+  if (type == "success"){
+    $(element_within).prepend("<div class='alert alert-success' role='alert'><i aria-hidden='true' class='fa fa-check-circle'></i>" + message + "<button aria-label='Dismiss alert' class='close' data-dismiss='alert'>&times;</button></div>");
+  }
+  else if (type = "error"){
+    $(element_within).prepend("<div class='alert alert-error' role='alert'><i aria-hidden='true' class='fa fa-times-circle'></i>" + message + "<button aria-label='Dismiss alert' class='close' data-dismiss='alert'>&times;</button></div>");
+  }
+  else if (type == "warning"){
+    $(element_within).prepend("<div class='alert alert-warning' role='alert'><i aria-hidden='true' class='fa fa-exclamation-triangle'></i>" + message + "<button aria-label='Dismiss alert' class='close' data-dismiss='alert'>&times;</button></div>");
+  }
+  else {
+    $(element_within).prepend("<div class='alert alert-warning' role='alert'><i aria-hidden='true' class='fa fa-info-circle'></i>" + message + "<button aria-label='Dismiss alert' class='close' data-dismiss='alert'>&times;</button></div>");
+  }
+  $(element_within).focus();
+  $(element_within).get(0).scrollIntoView({ block: "center", behavior: "smooth" });
+}
+
+/* ===== Make Screenreader-Only Alert ===== */
+function makeScreenreaderAlert(element_id, on_message, off_message){
+  let element = '#' + element_id;
+  if ($(element).length){
+    $(element).text(message);
+  }
+  else {
+    $('#screenreaderAlerts').apend('<div id="' + element_id + '" role="alert">' + message + '</div>');
+  }
+}
+
+/* ===================================== */
+/* ========= Clean JSON Parse ========== */
+/* ===================================== */
+function cleanJSON(json){
```

Comment:
What?!  Is this an attempt at pretty print?

Reply:
It is the only way I found to parse the JSON into a single line without unnecessary special character but also grammatically correct for sticklers and the screenreader-users.